pkg/controller/controlplane/valuesprovider.go

```diff
@@ -702,6 +707,66 @@ func (vp *valuesProvider) getControlPlaneShootChartValues(ctx context.Context, c
 		})
 	}
 
+	networkAccessType := apismetal.NetworkAccessBaseline
+	if cpConfig.NetworkAccessType != nil {
+		networkAccessType = *cpConfig.NetworkAccessType
+	}
+	restrictedOrForbidden := networkAccessType != apismetal.NetworkAccessBaseline
+	if restrictedOrForbidden && partition.NetworkIsolation == nil {
+		return nil, fmt.Errorf("cluster %s/%s control plane config with network access type %q requires network access to be set on partition", cluster.ObjectMeta.Namespace, cluster.ObjectMeta.Name, networkAccessType)
+	}
+
+	var dnsCidrs []string
+	if restrictedOrForbidden && partition.NetworkIsolation != nil {
+		dnsCidrs = make([]string, len(partition.NetworkIsolation.DNSServers))
+		for i, ip := range partition.NetworkIsolation.DNSServers {
+			parsedIP, err := netip.ParseAddr(ip)
+			if err != nil {
+				return nil, fmt.Errorf("unable to parse dns ip:%w", err)
+			}
+			if parsedIP.Is4() {
+				dnsCidrs[i] = ip + ipv4HostMask
+			}
+			if parsedIP.Is6() {
+				dnsCidrs[i] = ip + ipv6HostMask
+			}
+		}
+	}
+	if len(dnsCidrs) == 0 {
+		dnsCidrs = []string{ipv4Any}
+	}
```

Comment:
I am not sure but maybe this should only be done for baseline clusters?

Reply:
I think so, too. I added errors in case no dns or ntp is allowed. This should never happen.